infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/multiuser/oauth/OpenshiftTokenInitializationFilter.java

```diff
@@ -78,46 +79,43 @@ public OpenshiftTokenInitializationFilter(
   }
 
   @Override
-  protected String getUserId(String token) {
+  protected Optional<io.fabric8.openshift.api.model.User> processToken(String token) {
+    // We're effectively creating new client for each request. It might be a good idea to somehow
+    // cache the client or user object. However, it may require non-trivial refactoring which may
+    // be unnecessary. Keeping it as is for now to avoid premature optimization.
     try {
-      io.fabric8.openshift.api.model.User user = getCurrentUser(token);
-      return firstNonNull(user.getMetadata().getUid(), user.getMetadata().getName());
+      OpenShiftClient client = clientFactory.createAuthenticatedClient(token);
+      return Optional.of(client.currentUser());
```

Comment:
can `client.currentUser()` return null? If yes, you need to use `Optional.ofNullable`

Reply:
While the `Gettable.get()` method (used under the hood of the `currentUser()` is documented to possibly return `null`, I think here we have a special case, because we're looking up the special user `~` which corresponds to the currently logged in user. If there is no logged in user, we'd get the unauthorized exception, this theoretically should either return non-null or throw.

That said, let's be on the defensive side and use your suggestion :)